File: src/firstNonRepeatingCharacterInAStreamOfChars.py
import collections
class Solution:
    def solve(self, s):
        res = ""
        q = collections.deque()
        count = [0 for x in range(26)]
        for i in range(len(s)):
            q.append(s[i])
            count[ord(s[i])-ord('a')]+=1
            while q:
                if count[ord(q[0])-ord('a')] > 1:
                    q.popleft()
                else:
                    res+=q[0]
                    break

            if not q:
                res+="#"
        return res

File: src/firstNonRepeatingCharacterInAStreamOfChars.java
import java.util.LinkedList;
import java.util.Queue;

public class Solution {
    public String solve(String s) {
        StringBuilder res = new StringBuilder();
        int[] count = new int[26];
        Queue<Character> q =  new LinkedList<Character>();
        for(int i=0; i<s.length();i++){
            count[s.charAt(i)-'a'] +=1;
            q.add(s.charAt(i));

            while(!q.isEmpty()){
                if(count[q.peek()-'a'] > 1)
                    q.remove();
                else
                {
                    res.append(q.peek());
                    break;}
            }
            if(q.isEmpty()) res.append("#");
        }
        return res.toString();
    }
}
